### PR TITLE
JD5JB-41 - auto email feature

### PR DESCRIPTION
**Pull request by lukaszmarwitz**
added mailing feature to admin panel (just missing scheduler option)
updated dockerfile to auto copy of statistics.txt to wildfly directory
added logs to report generator and statistics manager

### CODE REVIEW

**Comment by isacademyci** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [ReportGenerator.java#L20](https://github.com/infoshareacademy/jjdd5-jbusters/blob/07b5d1c1da6ddc53073a91c1b6eeb2011b3126fe/app/src/main/java/com/infoshareacademy/jbusters/data/ReportGenerator.java#L20): Remove this unused import 'org.slf4j.Logger'. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [ReportGenerator.java#L21](https://github.com/infoshareacademy/jjdd5-jbusters/blob/07b5d1c1da6ddc53073a91c1b6eeb2011b3126fe/app/src/main/java/com/infoshareacademy/jbusters/data/ReportGenerator.java#L21): Remove this unused import 'org.slf4j.LoggerFactory'. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [StatisticsManager.java#L187](https://github.com/infoshareacademy/jjdd5-jbusters/blob/07b5d1c1da6ddc53073a91c1b6eeb2011b3126fe/app/src/main/java/com/infoshareacademy/jbusters/data/StatisticsManager.java#L187): Use a StringBuilder instead. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1643)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by isacademyci** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [StatisticsManager.java#L187](https://github.com/infoshareacademy/jjdd5-jbusters/blob/c029d3a7ddf8c942d841865f1baf74e9a71e8702/app/src/main/java/com/infoshareacademy/jbusters/data/StatisticsManager.java#L187): Use a StringBuilder instead. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1643)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: app.log [hunk 1]
```diff
@@ -241,3 +241,8 @@ Kategoria roku budowy:	2
 2018-12-13 15:07:33,612 INFO com.infoshareacademy.jbusters.data.Data [main] Load file CSV. Number of loaded transactions: 3363
 2018-12-13 15:07:34,087 INFO com.infoshareacademy.jbusters.data.DataLoader [main] Create list transaction. List size: 3363 rows
 2018-12-13 15:07:34,091 INFO com.infoshareacademy.jbusters.data.Data [main] district property file loaded successfully
```
Comment (MichalNowakowski) [inline app.log]: usun logi z repo pls

File: app/src/main/java/com/infoshareacademy/jbusters/data/MailHandler.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+	public void executor() throws UnsupportedEncodingException, MessagingException {
+
+		String name = SENDER_NAME;
+		String from = SENDER_USER_LOGIN;
+		String[] to = RECIPIENTS_LIST;
+		String subject = SUBJECT;
+		String body = CONTENT;
+		String attachmentPath = ATTACHMENT_PATH;
+		String attachmentName = ATTACHMENT_NAME;
```
Comment (MichalNowakowski) [inline app/src/main/java/com/infoshareacademy/jbusters/data/MailHandler.java]: po co przypiszujesz wartosci ze stalych do zmiennych zeby przekazac je do metody nizej? przekazuj stale od razu a jelsi metoda i tak nie pozwala na parametryzowanie tego, to moze lepiej w ogole wyrzucic te zmienne z metody?

File: app/src/main/java/com/infoshareacademy/jbusters/data/MailHandler.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+		sendFromGMail(name, from, to, subject, body, attachmentPath, attachmentName);
+	}
+
+	private static void sendFromGMail(String name, String from, String[] to, String subject, String body, String attachmentPath, String attachmentName) throws UnsupportedEncodingException, MessagingException {
+
+	    Properties props = System.getProperties();
+		String host = "smtp.gmail.com";
```
Comment (MichalNowakowski) [inline app/src/main/java/com/infoshareacademy/jbusters/data/MailHandler.java]: formatowanie do poprawy

File: web/src/main/webapp/WEB-INF/fm-templates/admin-panel.ftlh [hunk 4]
```diff
@@ -59,12 +59,39 @@
...
+                                </div>
+                            </div>
+                        </div>
+
+                        <div class="col-sm-6">
+                            <div class="card">
+                                <div class="card-body">
+                                    <h5 class="card-title">Opcja...</h5>
+                                    <p class="card-text">Tekst...</p>
+                                    <a href="/load-city" class="btn btn-primary">Przejdz dalej</a>
```
Comment (MichalNowakowski) [inline web/src/main/webapp/WEB-INF/fm-templates/admin-panel.ftlh]: polski znak - brakuje ź

**Comment by isacademyci** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [StatisticsManager.java#L187](https://github.com/infoshareacademy/jjdd5-jbusters/blob/99aff2a81428f5d70cbe4cacf69862412c741d0d/app/src/main/java/com/infoshareacademy/jbusters/data/StatisticsManager.java#L187): Use a StringBuilder instead. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1643)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**